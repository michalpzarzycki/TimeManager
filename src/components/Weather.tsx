import React, { useEffect, useState } from 'react';
import weatherService from '../services/weatherService'
import styles from './Weather.module.css'

interface IWeather  {
    city: string,
    temp: any, 
    icon: string
}
export default function Weather() {
    const [weather, setWeather] = useState<IWeather>({temp:'', city:'', icon:''})
    useEffect(() => {
            weatherService.getWeather().then((data: any) => {
                setWeather({city:data.name, icon:data.weather[0].icon, temp: data.main.temp})
            })
    }, [])
    useEffect(()=>{
        let iconElem = document.querySelector('#icon') as HTMLElement
        iconElem.style.backgroundImage = `url(http://openweathermap.org/img/w/${weather.icon}.png)`
    }, [weather])

    return(
        
            <div className={styles.mainDiv}>
                <div id='temp' className={styles.temp}>{weather.temp && (Math.round(weather.temp-273.15))}Â°C </div>
                 <div id='city' className={styles.city}>{weather.city && weather.city}</div>
                <div id='icon' className={styles.icon}>{}</div>
            </div>
    )
}