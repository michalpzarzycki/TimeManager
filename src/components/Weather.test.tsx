import React from 'react'
import {shallow} from 'enzyme'
import Weather from './Weather'
import {Provider} from 'react-redux'
import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});

test('Expext to render Weather component', () => {
    expect(shallow(<Provider store={store}><Weather choosenLocation="Warszaw"/></Provider>)).toMatchSnapshot()
})