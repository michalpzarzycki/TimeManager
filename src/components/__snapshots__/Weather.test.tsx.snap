// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expext to render Weather component 1`] = `ShallowWrapper {}`;
